presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -243,6 +252,10 @@ private synchronized void refreshNodesInternal()
         ImmutableSet.Builder<InternalNode> coordinatorsBuilder = ImmutableSet.builder();
         ImmutableSet.Builder<InternalNode> resourceManagersBuilder = ImmutableSet.builder();
         ImmutableSetMultimap.Builder<ConnectorId, InternalNode> byConnectorIdBuilder = ImmutableSetMultimap.builder();
+        Map<ConnectorId, SortedMap<String, InternalNode>> cachedConnectorNodes = new HashMap<>();
+        if (this.cachedNodesByConnectorId != null) {
+            cachedConnectorNodes.putAll(this.cachedNodesByConnectorId);
```

Comment:
Looks like cache will never expire because every time we will copy the already cached nodes. Offloading bin management to container management system is okay, I just want to make sure this is a conscious decision. And if this is the case, `cache` might not be a good name any more, and by having resilience enabled, we basically allow a node to have dead state, right?

Reply:
I agree. Removed cache from the name.